### ISSUE TITLE
Ceph: intermediately manually deleting persistent volume will not delete associated block image in ceph cluster

### ISSUE THREAD

**Issue by robinbraemer** [rook/rook#3491]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Associated block image remains in ceph pool

**Expected behavior:**
Delete the associated block image

**How to reproduce it (minimal and precise):**
Apply following yaml:
```yaml
apiVersion: ceph.rook.io/v1
kind: CephBlockPool
metadata:
  name: test
  namespace: rook-ceph
spec:
  replicated:
    size: 2
---
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
   name: test
provisioner: ceph.rook.io/block
parameters:
  blockPool: test
  clusterNamespace: rook-ceph
  # Specify the filesystem type of the volume. If not specified, it will use `ext4`.
  #fstype: xfs
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: mypv
spec:
  storageClassName: test
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 90Gi
---
apiVersion: v1
kind: Pod
metadata:
  name: mypod
spec:
  terminationGracePeriodSeconds: 0
  containers:
  - image: mysql:5.6
    name: mysql
    env:
    - name: MYSQL_ROOT_PASSWORD
      value: changeme
    ports:
    - containerPort: 3306
      name: mysql
    volumeMounts:
    - name: mysql-persistent-storage
      mountPath: /var/lib/mysql
  volumes:
  - name: mysql-persistent-storage
    persistentVolumeClaim:
      claimName: mypv
```
---------
**TL;DR**
```
kubectl delete pod mypod
copy pv name: kubectl get pv
kubectl delete pv <paste pv name>
Press CTRL+C
New terminal: kubectl -nrook-ceph logs -f pod/$(kubectl -nrook-ceph get pod -l "app=rook-ceph-operator" -o jsonpath='{.items[0].metadata.name}')
kubectl delete pvc mypv
see operator failing to delete peristent volume's block image
```
---------

All is fine an the rook-operator is creating the block image.

First delete the mypod pod.
`kubectl delete pod mypod`

Get and copy the created persistent volume that is bound to the mypv claim.
`kubectl get pv`

**Now following causes the "bug"...**
Delete the pv that you have copied.
`kubectl delete pv <paste pv name>`
Notice this command is not exiting as there is still is the mypv claim.
Press CTRL+C

Now when deleting the mypv claim rook-ceph-operator tries to delete the associated block image in the ceph pool but fails.
Watch the operator logs in a new terminal: `kubectl -nrook-ceph logs -f pod/$(kubectl -nrook-ceph get pod -l "app=rook-ceph-operator" -o jsonpath='{.items[0].metadata.name}')`
Delete the mypv claim: `kubectl delete pvc mypv`

The operator kicks off trying to delete the referenced pv failing due to exit code 16:
```
I0722 16:53:01.982983       6 controller.go:1333] delete "pvc-c7d6d30b-aca0-11e9-94ba-9600002bcd47": started
2019-07-22 16:53:01.986133 I | op-provisioner: Deleting volume pvc-c7d6d30b-aca0-11e9-94ba-9600002bcd47
2019-07-22 16:53:01.986200 I | exec: Running command: rbd rm test/pvc-c7d6d30b-aca0-11e9-94ba-9600002bcd47 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring
E0722 16:53:02.138774       6 controller.go:1356] delete "pvc-c7d6d30b-aca0-11e9-94ba-9600002bcd47": volume deletion failed: Failed to delete rook block image test/pvc-c7d6d30b-aca0-11e9-94ba-9600002bcd47: failed to delete image pvc-c7d6d30b-aca0-11e9-94ba-9600002bcd47 in pool test: Failed to complete '': exit status 16. 2019-07-22 16:53:02.115 7fbb0bfff700 -1 librbd::image::PreRemoveRequest: 0x55a33633aec0 check_image_watchers: image has watchers - not removing
Removing image: 0% complete...failed.
rbd: error: image still has watchers
This means the image is still open or the client using it crashed. Try again after closing/unmapping it or waiting 30s for the crashed client to timeout.
. output: 
W0722 16:53:02.138886       6 controller.go:937] Retrying syncing volume "pvc-c7d6d30b-aca0-11e9-94ba-9600002bcd47" because failures 3 < threshold 15
E0722 16:53:02.139303       6 controller.go:952] error syncing volume "pvc-c7d6d30b-aca0-11e9-94ba-9600002bcd47": Failed to delete rook block image test/pvc-c7d6d30b-aca0-11e9-94ba-9600002bcd47: failed to delete image pvc-c7d6d30b-aca0-11e9-94ba-9600002bcd47 in pool test: Failed to complete '': exit status 16. 2019-07-22 16:53:02.115 7fbb0bfff700 -1 librbd::image::PreRemoveRequest: 0x55a33633aec0 check_image_watchers: image has watchers - not removing
Removing image: 0% complete...failed.
rbd: error: image still has watchers
This means the image is still open or the client using it crashed. Try again after closing/unmapping it or waiting 30s for the crashed client to timeout.
. output: 
I0722 16:53:02.139382       6 event.go:209] Event(v1.ObjectReference{Kind:"PersistentVolume", Namespace:"", Name:"pvc-c7d6d30b-aca0-11e9-94ba-9600002bcd47", UID:"c7f40f2b-aca0-11e9-94ba-9600002bcd47", APIVersion:"v1", ResourceVersion:"12823", FieldPath:""}): type: 'Warning' reason: 'VolumeFailedDelete' Failed to delete rook block image test/pvc-c7d6d30b-aca0-11e9-94ba-9600002bcd47: failed to delete image pvc-c7d6d30b-aca0-11e9-94ba-9600002bcd47 in pool test: Failed to complete '': exit status 16. 2019-07-22 16:53:02.115 7fbb0bfff700 -1 librbd::image::PreRemoveRequest: 0x55a33633aec0 check_image_watchers: image has watchers - not removing
Removing image: 0% complete...failed.
rbd: error: image still has watchers
This means the image is still open or the client using it crashed. Try again after closing/unmapping it or waiting 30s for the crashed client to timeout.
```

Ceph error code 16 means the image is still open, but is should not as the mypod pod is obviously deleted. The important part to reproduce the bug is "`kubectl delete pv <paste pv name>` & Press CTRL+C" which I thinks puts the pv in a state where it no longer watched by the operator when the mypv claim is deleted.

**Environment**:
* OS (e.g. from /etc/os-release): Debian GNU/Linux 10 (buster)
* Kernel (e.g. `uname -a`): Linux debian-8gb-nbg1-1 4.19.0-5-amd64 #1 SMP Debian 4.19.37-5 (2019-06-19) x86_64 GNU/Linux
* Cloud provider or hardware configuration: self-hosted 
* Rook version: release v1.0.4
* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type: https://github.com/rancher/k3s
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:11:31Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.4-k3s.1", GitCommit:"ac992f3ef6c6d71e14366a1f65318e10eb3f4232", GitTreeState:"clean", BuildDate:"2019-07-14T07:04+00:00Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
```
* Storage backend status (`ceph health`): HEALTH_OK

**Comment by iExalt** [comment]
Getting the same error when manually deleting PV. For some reason, PVCs even with the reclaim policy of Delete will never delete their PVs, requiring manual cleanup which then leads to this bug.

**Comment by wdbasson** [comment]
Experiencing the same issue on `Rook v0.9.3` and `Ceph v13.2.6`. Unable to do manual cleanup and cluster is filling up.

**Comment by zzswang** [comment]
same issue

**Comment by dbarreda** [comment]
Same issue with 

Using 15.2.9 (357616cbf726abb779ca75a551e8d02568e15b17) 
Running on Openshift 4.6.22
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.0+d46d32f", GitCommit:"d46d32fbc3f0c35d2855a4589062aaf6335acd48", GitTreeState:"clean", BuildDate:"2021-03-11T01:16:36Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
